core/unit_test/TestDeviceAndThreads.py

```diff
@@ -0,0 +1,111 @@
+#  @HEADER
+#  ************************************************************************
+
+#  Kokkos v. 3.0
+#  Copyright (2020) National Technology & Engineering
+#  Solutions of Sandia, LLC (NTESS).
+
+#  Under the terms of Contract DE-NA0003525 with NTESS,
+#  the U.S. Government retains certain rights in this software.
+
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+
+#  1. Redistributions of source code must retain the above copyright
+#  notice, this list of conditions and the following disclaimer.
+
+#  2. Redistributions in binary form must reproduce the above copyright
+#  notice, this list of conditions and the following disclaimer in the
+#  documentation and/or other materials provided with the distribution.
+
+#  3. Neither the name of the Corporation nor the names of the
+#  contributors may be used to endorse or promote products derived from
+#  this software without specific prior written permission.
+
+#  THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+#  EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#  Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+
+#  ************************************************************************
+#  @HEADER
+
+import unittest
+import subprocess
+
+PREFIX = "$<TARGET_FILE_DIR:KokkosCore_UnitTest_DeviceAndThreads>"
+EXECUTABLE = "$<TARGET_FILE_NAME:KokkosCore_UnitTest_DeviceAndThreads>"
+COMMAND = "/".join([PREFIX, EXECUTABLE])
+
+
+def GetFlag(flag, *extra_args):
+    p = subprocess.run([COMMAND, flag, *extra_args], capture_output=True)
+    if p.returncode != 0:
+        raise Exception(p.stderr.decode("utf-8"))
+    return int(p.stdout)
+
+
+class KokkosInitializationTestCase(unittest.TestCase):
+    def test_num_threads(self):
+        max_threads = GetFlag("max_threads")
+        if max_threads == 1:
+            self.skipTest("no host parallel backend enabled")
+        for num_threads in range(1, max_threads + 1):
```

Comment:
Let not do every single one. just choose like 3 or so, say 1, 3 and max_threads (3 so we have a non-power of two)